pkg/querier/querier.go

```diff
@@ -97,30 +101,35 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor, ingesterMaxQueryLookback time.Duration) storage.Queryable {
+func NewQueryable(dq, cq storage.Queryable, distributor Distributor, cfg Config) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
-		cqr, err := cq.Querier(ctx, mint, maxt)
-		if err != nil {
-			return nil, err
-		}
-
 		q := querier{
-			queriers:    []storage.Querier{cqr},
 			distributor: distributor,
 			ctx:         ctx,
 			mint:        mint,
 			maxt:        maxt,
 		}
 
 		// Include ingester only if maxt is within ingesterMaxQueryLookback w.r.t. current time.
-		if ingesterMaxQueryLookback == 0 || maxt >= time.Now().Add(-ingesterMaxQueryLookback).UnixNano()/1e6 {
+		now := model.Now()
+		if cfg.IngesterMaxQueryLookback == 0 || maxt >= int64(now.Add(-cfg.IngesterMaxQueryLookback)) {
 			dqr, err := dq.Querier(ctx, mint, maxt)
 			if err != nil {
 				return nil, err
 			}
 			q.queriers = append(q.queriers, dqr)
 		}
 
+		// Include store only if mint is within StoreMinQueryLookback w.r.t current time.
+		if cfg.StoreMinQueryLookback == 0 || model.Time(mint).Before(now.Add(-cfg.StoreMinQueryLookback)) {
```

Comment:
`Before()` is `<`, while I think `<=` would be safer. Maybe we can just use `mint <= int64(now.Add(-cfg.StoreMinQueryLookback))` like you did for the `IngesterMaxQueryLookback `?

_If agree, same comment applies to `pkg/querier/unified_querier.go` too_

Reply:
I don't have a strong preference, so I changed it to as you've suggested.